migrations/upgrade.ts

```diff
@@ -0,0 +1,185 @@
+import { ethers } from "hardhat";
+import chalk from "chalk";
+import hre from "hardhat";
+import { SkaleABIFile, getContractKeyInAbiFile, encodeTransaction, concatTransactions, upgrade } from "@skalenetwork/upgrade-tools";
+import { getManifestAdmin } from "@openzeppelin/hardhat-upgrades/dist/admin";
+import { ProxyAdmin } from "@skalenetwork/upgrade-tools/dist/typechain-types";
+
+async function getMarionette(abi: SkaleABIFile) {
+    return ((await ethers.getContractFactory("Marionette")).attach(
+        abi[getContractKeyInAbiFile("Marionette") + "_address"] as string
+    ));
+}
+
+export async function getDeployedVersion(abi: SkaleABIFile) {
+    const marionette = await getMarionette(abi);
+    try {
+        return await marionette.version();
+    } catch {
+        console.log(chalk.red("Can't read deployed version"));
+    }
+}
+
+export async function setNewVersion(safeTransactions: string[], abi: SkaleABIFile, newVersion: string) {
+    const marionette = await getMarionette(abi);
+    safeTransactions.push(encodeTransaction(
+        0,
+        marionette.address,
+        0,
+        marionette.interface.encodeFunctionData(
+            "execute", 
+            [
+                marionette.address,
+                0,
+                marionette.interface.encodeFunctionData("setVersion", [newVersion])            ]
+        )
+    ));
+}
+
+async function main() {
+    await upgrade(
+        "marionette",
+        "1.0.0",
+        getDeployedVersion,
+        setNewVersion,
+        [],
+        ["Marionette"],
+        // async (safeTransactions, abi, contractManager) => {
+        async () => {
+            // deploy new contracts
+        },
+        async (safeTransactions, abi, _, safeMockAddress) => {
```

Comment:
`safeMock` is only for testing stuff. Not sure in should be presented in this production script.